Model/RemotePageRepository.php

```diff
@@ -41,44 +36,63 @@ class RemotePageRepository
      * @var CacheInterface
      */
     private $cache;
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
 
     public function __construct(
         ClientFactory $clientFactory,
-        Configuration $config,
         SiteRepositoryInterface $siteRepository,
         SearchCriteriaBuilder $searchCriteriaBuilder,
         StateInterface $cacheState,
-        CacheInterface $cache
+        CacheInterface $cache,
+        LoggerInterface $logger
     ) {
-        $this->client = $clientFactory->create();
-        $this->config = $config;
+        $this->clientFactory = $clientFactory;
         $this->siteRepository = $siteRepository;
         $this->searchCriteriaBuilder = $searchCriteriaBuilder;
         $this->cacheState = $cacheState;
         $this->cache = $cache;
+        $this->logger = $logger;
     }
 
+    /**
+     * @return array
+     * @throws LocalizedException
+     */
     public function getList(): array
     {
         $pages = [];
 
         foreach ($this->getSites() as $site) {
-            $baseUrl = trim($site->getBaseurl(), '/');
+            $client = $this->clientFactory->get($site->getBaseurl());
 
             try {
-                $response = $this->client->get($baseUrl . '/wp-json/wp/v2/pages');
-            } catch (ClientException $e) {
-                return [];
+                $response = $client->request('GET', '/wp-json/wp/v2/pages');
+                $pages[$site->getSiteId()]['name'] = $site->getName();
+                $pages[$site->getSiteId()]['id'] = $site->getSiteId();
+                $pages[$site->getSiteId()]['data'] = json_decode($response->getContent(), true);
+            } catch (ExceptionInterface $exception) {
+                $this->logger->error($exception->getMessage());
+
+                throw new LocalizedException(
+                    __('Something went wrong with requesting a Wordpress resource. See logs for more information.'),
+                    $exception,
+                    $exception->getCode()
+                );
             }
-
-            $pages[$site->getSiteId()]['name'] = $site->getName();
-            $pages[$site->getSiteId()]['id'] = $site->getSiteId();
-            $pages[$site->getSiteId()]['data'] = json_decode($response->getBody()->getContents(), true);
         }
 
         return $pages;
     }
 
+    /**
+     * @param int $siteId
+     * @param int $pageId
+     * @return array|null
+     * @throws LocalizedException
+     */
     public function get(int $siteId, int $pageId): ?array
```

Comment:
Je gebruikt strict typing en toch heb je comments toegevoegd, dat hadden we toch niet afgesproken?

Reply:
Exception!